precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/Navigator.kt

```diff
@@ -1,21 +1,25 @@
 package moe.tlaster.precompose.navigation
 
 import androidx.compose.runtime.Composable
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.viewmodel.compose.viewModel
 import kotlinx.coroutines.flow.map
-import moe.tlaster.precompose.lifecycle.LifecycleOwner
-import moe.tlaster.precompose.stateholder.SavedStateHolder
-import moe.tlaster.precompose.stateholder.StateHolder
-import moe.tlaster.precompose.stateholder.currentLocalStateHolder
 
 /**
  * Creates or returns an existing [Navigator] that controls the [NavHost].
  * @param name: Identify the navigator so you can have as many navigator instances as you need.
  * @return Returns an instance of Navigator.
  */
 @Composable
-fun rememberNavigator(name: String = ""): Navigator {
-    val stateHolder = currentLocalStateHolder
-    return stateHolder.getOrPut("${name}Navigator") {
+fun rememberNavigator(key: String? = null): Navigator {
+    val viewModel = viewModel<NavigatorViewModel>(key = key)
```

Comment:
This fails on iOS with the following error:
`Uncaught Kotlin exception: kotlin.UnsupportedOperationException: `Factory.create(String, CreationExtras)` is not implemented. You may need to override the method and provide a custom implementation. Note that using `Factory.create(String)` is not supported and considered an error.`

You need to provide a factory, something like this should work:
```
val viewModel = viewModel<NavigatorViewModel>(key = key) {
    NavigatorViewModel()
}
```